File: engine_py/tests/test_production_job_progress.py
def test_production_job_progress_and_state(make_world):
    world = make_world()
    e = world.spawn_entity()
    world.set_component(
        e,
        "ProductionJob",
        {
            "state": "pending",
            "progress": 0,
            "recipe": "wood_plank",
        },
    )

    progress = world.get_production_job_progress(e)
    assert progress == 0
    state = world.get_production_job_state(e)
    assert state == "pending"

    world.set_production_job_state(e, "in_progress")
    world.set_production_job_progress(e, 2)
    new_progress = world.get_production_job_progress(e)
    assert new_progress == 2
    new_state = world.get_production_job_state(e)
    assert new_state == "in_progress"

File: engine/scripts/lua/tests/test_production_job_progress.lua
local assert = require("assert")

local function test_production_job_progress_and_state()
	set_mode("colony")
	local e = spawn_entity()
	set_component(e, "ProductionJob", { state = "pending", progress = 0, recipe = "wood_plank" })

	local progress = get_production_job_progress(e)
	assert.equals(0, progress, "Initial production job progress should be 0")
	local state = get_production_job_state(e)
	assert.equals("pending", state, "Initial production job state should be 'pending'")

	set_production_job_state(e, "in_progress")
	set_production_job_progress(e, 2)
	local new_progress = get_production_job_progress(e)
	assert.equals(2, new_progress, "Production job progress should be 2 after set")
	local new_state = get_production_job_state(e)
	assert.equals("in_progress", new_state, "Production job state should be 'in_progress' after set")
end

return {
	test_production_job_progress_and_state = test_production_job_progress_and_state,
}
